x-pack/plugins/fleet/server/services/epm/packages/get.test.ts

```diff
@@ -0,0 +1,171 @@
+          updated_at: '2020-12-21T19:22:04.905Z',
+          version: 'WzIxNSwxXQ==',
+          score: 0,
+        },
+        {
+          type: 'ingest-package-policies',
+          id: 'dcf83172-c38e-4501-b236-9f479da8a7d6',
+          attributes: {
+            name: 'system-2',
+            description: '',
+            namespace: 'default',
+            policy_id: '11111-111111-11111-11111',
+            enabled: true,
+            output_id: '',
+            inputs: [],
+            package: { name: 'system', title: 'System', version: '0.10.4' },
+            revision: 1,
+            created_at: '2020-12-22T21:28:05.380Z',
+            created_by: 'elastic',
+            updated_at: '2020-12-22T21:28:05.380Z',
+            updated_by: 'elastic',
+          },
+          references: [],
+          migrationVersion: { 'ingest-package-policies': '7.11.0' },
+          updated_at: '2020-12-22T21:28:05.383Z',
+          version: 'WzE1NTAsMV0=',
+          score: 0,
+        },
+        {
+          type: 'ingest-package-policies',
+          id: 'dcf83172-c38e-4501-b236-9f479da8a7d6',
+          attributes: {
+            name: 'system-3',
+            description: '',
+            namespace: 'default',
+            policy_id: '22222-22222-2222-2222',
+            enabled: true,
+            output_id: '',
+            inputs: [],
+            package: { name: 'system', title: 'System', version: '0.10.4' },
+            revision: 1,
+            created_at: '2020-12-22T21:28:05.380Z',
+            created_by: 'elastic',
+            updated_at: '2020-12-22T21:28:05.380Z',
+            updated_by: 'elastic',
+          },
+          references: [],
+          migrationVersion: { 'ingest-package-policies': '7.11.0' },
+          updated_at: '2020-12-22T21:28:05.383Z',
+          version: 'WzE1NTAsMV0=',
+          score: 0,
+        },
+        {
+          type: 'ingest-package-policies',
+          id: 'dcf83172-c38e-4501-b236-9f479da8a7d6',
+          attributes: {
+            name: 'system-4',
+            description: '',
+            namespace: 'default',
+            policy_id: '33333-33333-333333-333333',
+            enabled: true,
+            output_id: '',
+            inputs: [],
+            package: { name: 'system', title: 'System', version: '0.10.4' },
+            revision: 1,
+            created_at: '2020-12-22T21:28:05.380Z',
+            created_by: 'elastic',
+            updated_at: '2020-12-22T21:28:05.380Z',
+            updated_by: 'elastic',
+          },
+          references: [],
+          migrationVersion: { 'ingest-package-policies': '7.11.0' },
+          updated_at: '2020-12-22T21:28:05.383Z',
+          version: 'WzE1NTAsMV0=',
+          score: 0,
+        },
+      ];
+      soClient.find.mockImplementation(async ({ page = 1, perPage = 20 }) => {
+        let savedObjectsResponse: typeof savedObjects;
+
+        switch (page) {
+          case 1:
```

Comment:
so in this mock.. if it's page 1, you return index 1, the second element.

If it's page 2, you return indices 1,2,3?

It's not clear why you're doing this. It also doesn't seem like any of these scenarios are necessary in the tests below.

unless this is strictly for making sure the `getPackageUsageSummary` call doesn't error, but then I think this could all be simpler

Reply:
Good catch. This here (below) should be `savedObjects[0]`.
the idea is: 
- page 1 - return first item in array
- page 2 - return every other array item after the first one

Doing this ensures that paging through the data works and that it de-dups the packages that are assigned to the same agent policy